### DISCUSSION TITLE
getting bad request for https://openapi.etsy.com/v3/application/shops/{shop_id}/receipts

### DISCUSSION THREAD

**Discussion by Ahmedkhan22** [etsy/open-api#1087] [Q&A]
am sending request and in response i am getting bad request i dont know why i checked the scope permissions still cant find anything wrong in my code

**Comment by emergingdzns** [comment]
Can you post the full response? What is the error code? Maybe post your full payload (minus the token and api key) and we may be able to help.

**Reply by Ahmedkhan22** [reply]
var Full_URL=`https://www.etsy.com/oauth/connect?response_type=code&redirect_uri=http://localhost:3003/oauth/redirect&scope=transactions_r%20email_r&client_id=${clientID}&state=${state}&code_challenge=${codeChallenge}&code_challenge_method=S256`;
thats how i get the token
below is another api req that api hit after i get the token
const requestOptions = {
  headers: {
      'x-api-key': apiKey,
      // Scoped endpoints require a bearer token
      Authorization: `Bearer ${accessToken}`,
  }
};

  const url = `https://api.etsy.com/v3/application/shops/${shopId}/receipts`
  const response = await fetch(`${url}`, {
      headers: {
        Authorization: `Bearer ${accessToken}`,
        "x-api-key":clientID
      },
    });
    below is the error
    Failed to fetch shop receipts: Response {
  [Symbol(realm)]: null,
  [Symbol(state)]: {
    aborted: false,
    rangeRequested: false,
    timingAllowPassed: true,
    requestIncludesCredentials: true,
    type: 'default',
    status: 403,
    timingInfo: {
      startTime: 292368.5312000513,
      redirectStartTime: 0,
      redirectEndTime: 0,
      postRedirectStartTime: 292368.5312000513,
      finalServiceWorkerStartTime: 0,
      finalNetworkResponseStartTime: 0,
      finalNetworkRequestStartTime: 0,
      endTime: 0,
      encodedBodySize: 45,
      decodedBodySize: 45,
      finalConnectionTimingInfo: null
    },
    cacheState: '',
    statusText: 'Forbidden',
    headersList: HeadersList {
      cookies: [Array],
      [Symbol(headers map)]: [Map],
      [Symbol(headers map sorted)]: null
    },
    urlList: [ [URL] ],
    body: { stream: undefined }
  },
  [Symbol(headers)]: HeadersList {
    cookies: [
      'user_prefs=ZuHMQYs3pVdw-EfQSn7ISgiuPR9jZACClHsJK2B0dF5pTo4OeUQsAwA.; expires=Fri, 16 Aug 2024 18:02:16 GMT; Max-Age=31536000; path=/; domain=.etsy.com'
    ],
    [Symbol(headers map)]: Map(11) {
      'server' => [Object],
      'x-etsy-request-uuid' => [Object],
      'x-limit-per-second' => [Object],
      'x-remaining-this-second' => [Object],
      'x-limit-per-day' => [Object],
      'x-remaining-today' => [Object],
      'content-length' => [Object],
      'content-type' => [Object],
      'date' => [Object],
      'connection' => [Object],
      'set-cookie' => [Object]
    },
    [Symbol(headers map sorted)]: null
  }
}

**Reply by seandmccarthy** [reply]
Do other API requests work ok for you? What do you get if you try `getShop`? 
https://developers.etsy.com/documentation/reference/#operation/getShop

**Reply by Ahmedkhan22** [reply]
@seandmccarthy   const requestOptions = {
      headers: {
          'x-api-key': clientID,
          // Scoped endpoints require a bearer token
          Authorization: `Bearer ${access_token}`,
      }
  };

  const response = await fetch(
      `https://api.etsy.com/v3/application/users/${user_id}`,
      requestOptions
  );
  this API work ok for me

**Reply by seandmccarthy** [reply]
How did you get the `shop_id` you used in your original request to get receipts?

If you call https://developers.etsy.com/documentation/reference/#operation/getShop
with that `shop_id` do you get a valid response?

What about calling https://developers.etsy.com/documentation/reference/#operation/getShopByOwnerUserId with the `user_id` in your successful request?

**Reply by Ahmedkhan22** [reply]
@seandmccarthy i get the shop id from the url which shows when you place your cursor at chat or contact us

**Reply by seandmccarthy** [reply]
What do you get back from the `getShop` API call with that shop_id ?
